lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -120,14 +147,44 @@ public StorageCandidate(CubeInterface cube, CubeFactTable fact, String storageNa
       .getFloat(CubeQueryConfUtil.COMPLETENESS_THRESHOLD, CubeQueryConfUtil.DEFAULT_COMPLETENESS_THRESHOLD);
   }
 
-  @Override
-  public String toHQL() {
-    return null;
+  public StorageCandidate(StorageCandidate sc) {
+    this(sc.getCube(), sc.getFact(), sc.getStorageName(), sc.getAlias(), sc.getCubeql());
+    // Copy update periods.
+    for (UpdatePeriod updatePeriod : sc.getValidUpdatePeriods()) {
+      this.validUpdatePeriods.add(updatePeriod);
+    }
+  }
+
+  static boolean containsAny(Collection<String> srcSet, Collection<String> colSet) {
```

Comment:
Taken care and moved to CandaiateUtil
`  public static boolean containsAny(Set<String> srcSet, Set<String> colSet) {
    if (colSet == null || colSet.isEmpty()) {
      return true;
    }
    for (String column : colSet) {
      if (srcSet.contains(column)) {
        return true;
      }
    }
    return false;
  }`